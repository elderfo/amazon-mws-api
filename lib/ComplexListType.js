/**
 * Complex List helper object. Once initialized, you should set
 * an add(args) method which pushes a new complex object to members.
 *
 * @param {String} name Name of Complex Type (including .member or subtype)
 */
export default class ComplexListType {
  constructor(name) {
    this.pre = name;
    this.members = [];
  }

  /**
   * Appends each member object as a complex list item
   * @param  {Object} query Query object to append to
   * @return {Object}       query
   */
  appendTo(query) {
    const workingQuery = { ...query };

    this.members.forEach((member, idx) => {
      Object.keys(member).forEach((key) => {
        workingQuery[`${this.pre}.${idx + 1}.${key}`] = member[key];
      });
    });

    return workingQuery;
  }
}
