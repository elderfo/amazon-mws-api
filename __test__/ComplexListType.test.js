import ComplexListType from '../lib/ComplexListType';

describe('ComplexList', () => {
  test('should not be null', () => {
    expect(ComplexListType).toBeDefined();
  });

  test('should initialize with expected values', () => {
    const complexListType = new ComplexListType('newName');
    expect(complexListType).toMatchSnapshot();
  });

  test('appendTo should perform expected work', () => {
    const complexListType = new ComplexListType('newName');

    complexListType.members.push({ abc: '123' });

    const query = {};
    const actual = complexListType.appendTo(query);

    expect(actual).toMatchSnapshot();
  });
});
