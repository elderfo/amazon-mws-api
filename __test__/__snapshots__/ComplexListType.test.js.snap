// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComplexList appendTo should perform expected work 1`] = `
Object {
  "newName.1.abc": "123",
}
`;

exports[`ComplexList should initialize with expected values 1`] = `
ComplexListType {
  "members": Array [],
  "pre": "newName",
}
`;
